modules/sdp-simulcast/index.js

```diff
@@ -0,0 +1,510 @@
+        if (fidGroup) {
+            primarySsrc = parseInt(fidGroup.ssrcs.split(" ")[0]);
+        } else {
+            // Unsupported scenario
+            return mLine;
+        }
+    }
+    console.log("SdpSimulcast: current ssrc cache: ", this.ssrcCache);
+    console.log("SdpSimulcast: parsed primary ssrc " + primarySsrc);
+
+    var seenPrimarySsrc = this.ssrcCache.indexOf(primarySsrc) !== -1;
+
+    if (seenPrimarySsrc) {
+        console.log("SdpSimulcast: Have seen primary ssrc before, " +
+            "filling in data from cache");
+        mLine = this._fillInSourceDataFromCache(mLine);
+    } else {
+        console.log("SdpSimulcast: Have not seen primary ssrc before, " +
+            "generating source data");
+        mLine = this._generateSourceData(mLine, primarySsrc);
+    }
+    // Now update the cache to match whatever we've just put into this sdp
+    this.ssrcCache = this._parseSimLayers(mLine);
+    return mLine;
+}
+
+//endregion
+
+//region "Public" functions
+
+/**
+ *
+ * @param desc
+ * @param enableConferenceFlag
+ * @returns {RTCSessionDescription}
+ */
+Simulcast.prototype.mungeRemoteDescription = function (desc, enableConferenceFlag) {
+
+    if (!validateDescription(desc)) {
+        return desc;
+    }
+
+    var session = transform.parse(desc.sdp);
+
+    var self = this;
+    processVideo(session, function (mLine) {
+
+        // Handle simulcast reception.
+        if (self.options.explodeRemoteSimulcast) {
+            explodeRemoteSimulcast(mLine);
+        } else {
+            implodeRemoteSimulcast(mLine);
+        }
+
+        // Add or remove "x-google-conference" from the remote description based on whether the client
+        // has enabled simulcast for the local video source. For cases where we disable simulcast for desktop share,
+        // it is necessary to remove the flag so that Chrome stops sending T1 temporal layers. It also fixes other
+        // issues related to screensharing like https://bugs.chromium.org/p/chromium/issues/detail?id=1093819.
+        if (!self.options.usesUnifiedPlan && enableConferenceFlag) {
+            assertGoogConference(mLine);
+        } else {
+            removeGoogConference(mLine);
+        }
+    });
+
+    return new RTCSessionDescription({
+        type: desc.type,
+        sdp: transform.write(session)
+    });
+};
+
+/**
+ *
+ * NOTE this method should be called only if simulcast is supported by
+ * the current browser, otherwise local SDP should not be munged.
+ * @param desc
+ * @returns {RTCSessionDescription}
+ */
+Simulcast.prototype.mungeLocalDescription = function (desc) {
+
+    if (!validateDescription(desc)) {
+        return desc;
+    }
+
+    var session = transform.parse(desc.sdp);
+
+    var self = this;
+    processVideo(session, function (mLine) {
+        if (mLine.direction == 'recvonly' || mLine.direction == 'inactive' || mLine.mid == '3')
```

Comment:
another part of POC magic: prevents from adding simulcast for desktop track added on the 3rd mline